docs/components/table.md

```diff
@@ -123,58 +73,142 @@ See [the State section](#error) about `hasError`
 
 See [the State section](#loading) about `isLoading`
 
+### fetcher
+
+Use a custom fetcher function to fetch table data and leverage server-side search, sort and pagination.
+::: tip Note: 
+All fetcher functions should take a single param. This parameter is a JSON
+object supporting the following properties: 
+  - Pagination support: `page`, `pageSize`
+  - Sort support: `sortColumnKey`, `sortColumnOrder`
+  - Search support: `query`
+:::
+
+::: tip Note: 
+All fetcher functions should return a JSON object. This JSON object should contain the following properties: 
+  - `total` - the total count of items (if using pagination)
+  - `data` - an array of JSON objects to populate the table with
+:::
+
+Example fetcher function:
+
+```js
+fetcher(payload) {
+  const params = {
+    _limit: payload.pageSize,
+    _page: payload.page
+  }
+
+  if (query) {
+    params.q = payload.query
+    params._page = 1
+  }
+
+  if (sortKey) {
+    params._sort = payload.sortColumnKey
+    params._order = payload.sortColumnOrder
+  }
+
+  return axios.get('/user_list', {
+    baseURL: 'https://kongponents.dev/api',
+    params
+  }).then(res => {
+    return {
+      total: res.total,
+      data: res.data
+    }
+  })
+}
+```
+
+### searchInput
+
+Pass in a string of search input for server-side table filtering. See [the Server-side function section](#server-side-functions) 
+for an example.
+
+
+### headers
+
+Pass in an array of header objects for the table.
+
+| Parameter | Type | Description
+|:-------- |:------- |:-------
+| `key` | string | A unique key for the column
+| `label` | string | The label displayed on the table for the column
+| `sortable` | boolean | Enables or disables server-side sorting for the column
```

Comment:
Would it make sense to include the default values here? e.g. `sortable` is `true/false` by default. Same for `hideLabel`, etc.

Reply:
Done!

![image](https://user-images.githubusercontent.com/67973710/141529158-5b21c791-acc2-4073-923c-ea35b6094bfc.png)